datastream-server/src/main/java/com/linkedin/datastream/server/assignment/StickyPartitionAssignmentStrategy.java

```diff
@@ -345,6 +444,55 @@ public StickyPartitionAssignmentStrategy(Optional<Integer> maxTasks, Optional<In
     return tasksToCleanUp;
   }
 
+  @Override
+  protected int constructExpectedNumberOfTasks(DatastreamGroup dg, List<String> instances,
+      Map<String, Set<DatastreamTask>> currentAssignmentCopy) {
+    // Count the number of tasks present for this datastream
+    Set<DatastreamTask> allAliveTasks = new HashSet<>();
```

Comment:
You can move this calculation in the else block of numTasks == 0, to avoid extra calculations.

Reply:
done